### PR TITLE
Update the sprint branch again with develop changes mostly for BUFR (via PR1100)

### PR DESCRIPTION
**Pull request by gthompsnJCSDA**
## Description

This PR again updates the top-level sprint branch to incorporate changes from develop (I think almost entirely PR1100).  With this update, we should be back on track to fixing the broken ctests related to naming convention changes so we can move ahead with the big push to get all sprint branches (UFO and IODA and all associates) to merge into develop in the next 2 weeks.  There will be about 15 ctests not yet working, mostly BUFR - continuing work in another branch.

### Issue(s) addressed

A separate one was not created.

## Acceptance Criteria (Definition of Done)

Fixing as many ctests as possible.

## Dependencies

There is a corresponding PR in ioda repo for the validation/ObsSpace.yaml
[ioda PR 868](https://github.com/JCSDA-internal/ioda/pull/868)

## Impact

Keeping in sprint branch until ready for merge to develop.

### CODE REVIEW

**Comment by gthompsnJCSDA** [comment]
Making this PR a draft until fixing a bunch of the 92 failed ctests.  Do not review yet.

File: src/gsi-ncdiag/gsi_ncdiag.py
Comment (BenjaminRuston) [inline src/gsi-ncdiag/gsi_ncdiag.py:140]: believe the left hand side is the `GSI` variable name so should remain as `LaunchTime` but am not entirely sure

**Review by BenjaminRuston — APPROVED** [review]
this looks like it is needed!   should get this in ASAP I would assume

**Comment by BenjaminRuston** [comment]
@gthompsnJCSDA this is quite large, is there anyway to break it up and get a part of it in now

**Review by srherbener — APPROVED** [review]
Great progress - down to 15 test failures on all three compilers. The rest of the tests can be addressed in subsequent PRs. Thanks for all your hard work to get this accompilished!

**Comment by srherbener** [comment]
@BenjaminRuston did you mean to reset your review?

Given that we will accept with test failures for this PR, is this ready to merge now? The companion PR in ioda is ready. thanks!

**Review by jeromebarre — CHANGES_REQUESTED** [review]
I think that the changes proposed in the tropomi conveter will produce files that won't work in UFO and this cannot be done at the moment due to constrains in IODA.

**Comment by srherbener** [comment]
> I think that the changes proposed in the tropomi conveter will produce files that won't work in UFO and this cannot be done at the moment due to constrains in IODA. Same goes with MOPITT CO but changes haven't been made or at least I haven't seen it.

@jeromebarre is this referring to the averaging kernel variables that cannot be 2D yet. That is, they need to remain as a set of 1D variables with the level (?) number attached as a suffix. If so, which specific sets of variables need to remain 1D? Thanks!

**Comment by srherbener** [comment]
@jeromebarre I've been informed from the OBS team that the agreed upon plan is to merge this PR as is, and revert the tropomi converter to the current develop version if necessary (before the Fortran API shows up).

Is that your understanding? Are you okay with merging as is?

**Comment by jeromebarre** [comment]
Yes this refers to the averaging kernel and pressure vertices arrays and other ancillary variables such as the apriori quantities. Merging the tropomi converter as is as this will produce useless IODA files, i.e. breaking the converter. We will most likely have to revert most of the changes made but at this point it now feels that we ought to merge as is otherwise we'll waste even more time and energy on this. Thanks for checking anyway.

**Comment by srherbener** [comment]
@jeromebarre thank you for you generous patience and understanding.

**Comment by srherbener** [comment]
We still need another reviewer approval before merging.

**Pull request merged**
